abci/types/types.proto

```diff
@@ -81,8 +81,15 @@ message RequestBeginBlock {
   repeated Evidence byzantine_validators = 4 [(gogoproto.nullable)=false];
 }
 
+enum CheckTxType {
+  Unchecked = 0;
```

Comment:
I'd call these `New` and `Recheck`. Having `CheckTxRequest` with type `Checked` is confusing to me.

Reply:
Renamed it as such 👍